src/k2/cmd/tpcc/transactions.h

```diff
@@ -800,3 +800,369 @@ class OrderStatusT : public TPCCTxn
+
+                return _txn.query(query_cid)
+                .then([this, idx] (auto&& response) {
+                    CHECK_READ_STATUS(response);
+
+                    // get customer ID
+                    dto::SKVRecord& rec = response.records[0];
+                    std::optional<int16_t> lineCountOpt = rec.deserializeField<int16_t>("LineCount");
+                    std::optional<int32_t> cidOpt = rec.deserializeField<int32_t>("CID");
+                    _o_line_count[idx] = *lineCountOpt;
+                    _O_C_ID[idx] = *cidOpt;
+
+                    K2DEBUG("[DeliveryT getCustomerID] _o_line_count(" << idx << ")=" << _o_line_count[idx]
+                            << ", _O_C_ID(" << idx << ")=" << _O_C_ID[idx]);
+                    return make_ready_future();
+                });
+            });
+        });
+    }
+
+    // update the carrierID in ORDER table
+    future<> updateCarrierID(uint32_t idx) {
+        std::vector<String> updateFields({"CarrierID"});
+
+        Order updateOrder(_w_id, _d_id[idx], _NO_O_ID[idx]);
+        updateOrder.CarrierID = _o_carrier_id[idx];
+
+        return partialUpdateRow<Order, std::vector<String>>(updateOrder, updateFields, _txn).discard_result();
+    }
+
+    // get the sum of all OL_AMOUNT in the ORDER_LINE table with mathing w_id, d_id, o_id
+    future<> getOLSumAmount(uint32_t idx) {
+        return _client.createQuery(tpccCollectionName, "orderline")
+        .then([this, idx](auto&& response) mutable {
+            CHECK_READ_STATUS(response);
+    
+            return do_with(std::move(response.query), [this, idx] (Query& query_order_line) {
+                // make Query request and set query rules. 
+                // 1) add Projection fields (i.e. amount) as needed;
+                query_order_line.startScanRecord.serializeNext<int16_t>(_w_id); 
+                query_order_line.startScanRecord.serializeNext<int16_t>(_d_id[idx]);
+                query_order_line.startScanRecord.serializeNext<int64_t>(_NO_O_ID[idx]);
+                query_order_line.startScanRecord.serializeNext<int32_t>(1);
+                query_order_line.endScanRecord.serializeNext<int16_t>(_w_id);
+                query_order_line.endScanRecord.serializeNext<int16_t>(_d_id[idx]);
+                query_order_line.endScanRecord.serializeNext<int64_t>(_NO_O_ID[idx]);
+                query_order_line.endScanRecord.serializeNext<int32_t>(_o_line_count[idx] + 1);
+                query_order_line.setLimit(_o_line_count[idx]);
+                query_order_line.setReverseDirection(false);
+
+                std::vector<String> projection{"Amount"};   // make projection
+                query_order_line.addProjection(projection);
+                dto::expression::Expression filter{};       // make filter Expression
+                query_order_line.setFilterExpression(std::move(filter));
+
+                return _txn.query(query_order_line)
```

Comment:
You need to call query until "isDone" is true so that you handle paginated responses.